ide/src/main/java/org/aya/ide/action/FindReferences.java

```diff
@@ -31,6 +32,22 @@ public interface FindReferences {
     });
   }
 
+  static @NotNull SeqView<SourcePos> findRefsOutsideDefs(
+    @NotNull LibrarySource source,
+    @NotNull SeqView<LibraryOwner> libraries, XY xy
+  ) {
+    var vars = Resolver.resolveVar(source, xy);
+    return findRefsOutsideDefs(vars.map(WithPos::data), libraries);
+  }
+
+  static @NotNull SeqView<SourcePos> findRefsOutsideDefs(
+    @NotNull SeqView<AnyVar> vars,
+    @NotNull SeqView<LibraryOwner> libraries
+  ) {
+    var defPos = vars.filterIsInstance(DefVar.class).map(def -> def.concrete.entireSourcePos());
+    return findRefs(vars, libraries).filter(ref -> !defPos.map(pos -> pos.containsIndex(ref)).contains(true));
```

Comment:
I wonder if this would do extra work because Java is not lazy.

Reply:
What do you mean by "extra work"?